packages/client/lib/service/fullethereumservice.ts

```diff
@@ -190,9 +191,18 @@ export class FullEthereumService extends EthereumService {
       for (const hash of hashes) {
         const blockReceipts = await receiptsManager.getReceipts(hash, true, true)
         if (!blockReceipts) continue
-        blockReceipts.forEach((r) => (r.gasUsed = bigIntToBuffer(r.gasUsed) as any))
         receipts.push(...blockReceipts)
-        receiptsSize += Buffer.byteLength(JSON.stringify(blockReceipts))
+        const receiptsBuffer = Buffer.concat(
+          receipts.map((r) =>
+            rlp.encode([
+              (r as PreByzantiumTxReceipt).stateRoot ?? (r as PostByzantiumTxReceipt).status,
+              bigIntToBuffer(r.gasUsed),
+              r.bitvector,
+              r.logs,
+            ])
+          )
+        )
+        receiptsSize += Buffer.byteLength(receiptsBuffer)
```

Comment:
Could you drop a short note what's happening here?

Does this fix some concrete issues? Can't oversee this on a first look.

Reply:
ah yeah, actually you commented on an old version of this code, if you look at the final version in this PR i made it much more simple by using our existing encodeReceipt function. this is simply keeping track of the size of the GetReceipts response, so we can respect capping at 10mib as the spec suggests.